src/DCDataSet.cpp

```diff
@@ -34,14 +36,60 @@
 #include "splash/DCException.hpp"
 #include "splash/basetypes/ColTypeDim.hpp"
 
+herr_t H5Ewalk_strings(unsigned n, const H5E_error2_t *err_desc, void *client_data)
+{
+    //FILE		*stream = (FILE *)client_data;
+    std::ostringstream *stream = (std::ostringstream *)client_data;
```

Comment:
I think you can use the normal object of `ostringstream` here, no need to use the pointer.

just pass the stringstream by reference to the function.